src/main/java/calculator/StringCalculator.java

```diff
@@ -0,0 +1,64 @@
+package calculator;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringCalculator {
+
+    public static final int ZERO = 0;
+    public static final String DEFAULT_DELIMITER = ",|:";
+    public static final String CUSTOM_DELIMITER_FIND_REGEX = "//(.*)\n(.*)";
+
+    public static int splitAndSum(String text) {
+        int sum = ZERO;
+        if (isBlank(text)) {
+            return sum;
+        }
+
+        sum = sum(getNumbers(split(text)));
+        return sum;
+    }
+
+    private static boolean isBlank(String text) {
+        return text == null || text.isBlank();
+    }
+
+    private static int sum(int[] numbers) {
+        int sum = ZERO;
+        for (int number : numbers) {
+            sum += number;
+        }
+        return sum;
+    }
+
+    private static int[] getNumbers(String[] stringNumbers) {
+        int[] numbers = new int[stringNumbers.length];
+        for (int i = 0; i < stringNumbers.length; i++) {
+            numbers[i] = getPositiveNumber(new PositiveNumber(stringNumbers[i]));
+        }
+        return numbers;
+    }
```

Comment:
```suggestion
    private static List<Integer> getNumbers(String[] stringNumbers) {
        return Arrays.stream(stringNumbers)
                .map(PositiveNumber::new)
                .map(PositiveNumber::valueOf)
                .collect(Collectors.toList());
    }
```
- Stream API를 적극적으로 활용해보는건 어떨까요?
- getPositiveNumber 를 굳이 메소드로 분리할 필요가 있을까요? 위 예시에선 인터페이스를 유사하게 맞추기 위해 값을 꺼냈지만, PositiveNumber를 반환한 후, sum메소드도 PositiveNumber로 수행하는건 어떨까요?

Reply:
포비님께서 3단계전까지는 최대한 스트림사용을 지양하라고 하여 최대한 안쓰려고했는데
그럴께 아니라 둘다 하면 좋을꺼같네요 ㅎㅎㅎ
위에 지적해주신 sum 과 비슷하게 리팩토링을 진행하면서 필요없어지는 변수나 메소드가 있는지 잘 체크해야겠네요 ㅎㅎㅎ

Reply:
아, 포비가 스트림 지양하라고 하셨었군요 ㅎㅎ